File: Snake_Java/src/snake/Food.java
package snake;

import javax.swing.*;
import java.awt.*;

/**
 * Food.java
 *
 * Contains attributes and methods for drawing
 * and displaying of food.
 */
public class Food extends Component implements Drawable
{
    private int xPos, yPos;
    private static final String foodImg = "images/donut.png";

    public Food(int xPos, int yPos)
    {
        this.xPos = xPos;
        this.yPos = yPos;
    }

    /**
     * Draws an image of a donut
     *
     * @param g: Graphics object
     */
    public void draw(Graphics g)
    {
        ImageIcon img = new ImageIcon(foodImg);
        img.paintIcon(this, g, xPos, yPos);
    }

    /**
     * @return x position of food
     */
    public int getXPos()
    {
        return xPos;
    }

    /**
     * @param xPos: x position to set
     */
    public void setXPos(int xPos)
    {
        this.xPos = xPos;
    }

    /**
     * @return y position of food
     */
    public int getYPos()
    {
        return yPos;
    }

    /**
     * @param yPos: y position to set
     */
    public void setYPos(int yPos)
    {
        this.yPos = yPos;
    }
}

File: Snake_C++/src/Engine/Food.cpp
#include "../../include/Food.hpp"
#include "../../include/Settings.hpp"

using namespace Engine;
Food::Food() = default;

Food::Food(std::shared_ptr<Context> &context, int xPos, int yPos)
    : context(context), xPos(xPos), yPos(yPos)
{
}

/**
 * Initialise food image file and draws it to the current position
 */
void Food::draw()
{
    context->assets->addTexture(FOOD, "../assets/images/basketball.png");
    food.setTexture(context->assets->getTexture(FOOD));
    food.setPosition((float)xPos, (float)yPos);
    context->window->draw(food);
}

/**
 * @return x position of food
 */
int Food::getXPos()
{
    return xPos;
}

/**
 * @param x: x position to set
 */
void Food::setXPos(int x)
{
    this->xPos = x;
}

/**
 * @return y position of food
 */
int Food::getYPos()
{
    return yPos;
}

/**
 * @param y: y position to set
 */
void Food::setYPos(int y)
{
    this->yPos = y;
}
